tests/data/test_entryexitanalysis.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from unittest.mock import MagicMock, PropertyMock
+
+import pandas as pd
+import pytest
+
+from freqtrade.commands.analyze_commands import start_analysis_entries_exits
+from freqtrade.commands.optimize_commands import start_backtesting
+from freqtrade.enums import ExitType
+from freqtrade.optimize.backtesting import Backtesting
+from tests.conftest import get_args, patch_exchange, patched_configuration_load_config_file
+
+
+@pytest.fixture(autouse=True)
+def entryexitanalysis_cleanup() -> None:
+    yield None
+
+    Backtesting.cleanup()
+
+
+def test_backtest_analysis_nomock(default_conf, mocker, caplog, testdatadir, tmpdir, capsys):
+    default_conf.update({
+        "use_exit_signal": True,
+        "exit_profit_only": False,
+        "exit_profit_offset": 0.0,
+        "ignore_roi_if_entry_signal": False,
+        'analysis_groups': "0",
+        'enter_reason_list': "all",
+        'exit_reason_list': "all",
+        'indicator_list': "rsi"
+    })
```

Comment:
looking at the coverage - we should either have multiple tests (with different analysis_groups) - or provide more groups in this test (or parametrize it to use different combinations - but i suspect then asserting may become difficult - but maybe it's not? ).

Reply:
That might be a bit pointless with the current simplicity of the trades in the backtest. I'd have to add more trades. It's possible it'll just take me a bit longer to do.

Reply:
Done

Reply:
to be honest - that's also changed with the simplification of the whole method ...